src/main/java/com/epam/processor/DataProcessor.java

```diff
@@ -40,7 +53,16 @@ public RoadAccident getAccidentByIndex7(String index){
      * @return
      */
     public Collection<RoadAccident> getAccidentsByLocation7(float minLongitude, float maxLongitude, float minLatitude, float maxLatitude){
-        return null;
+        Collection<RoadAccident> filteredBylocation= new ArrayList<RoadAccident>();
+        for(RoadAccident roadAccident:roadAccidentList){
+            float longitude = roadAccident.getLongitude();
+            float latitude = roadAccident.getLatitude();
+            if(longitude>=minLongitude && longitude<=maxLongitude && latitude>=minLatitude && latitude<=maxLatitude) {
```

Comment:
I would recommend to extract such complicated logic in separated method with some meaningful name. If some other new-joiner look at this code, he will take some time to understand what it's checking. 

In contrast, if you are using method isInScope, like in [this](https://github.com/bagedan/EPAM-JMP/pull/1/files#diff-9566c4a7c6afefe7dea1a80cbc457ad8R53) PR, it makes more sense from the very first glance. Right? :smile: